eras/conway/impl/testlib/Test/Cardano/Ledger/Conway/Imp/RatifySpec.hs

```diff
@@ -35,6 +35,117 @@ spec =
     votingSpec
     delayingActionsSpec
     spoVotesForHardForkInitiation
+    paramChangeAffectsProposalsSpec
+
+paramChangeAffectsProposalsSpec ::
+  forall era.
+  ConwayEraImp era =>
+  SpecWith (ImpTestState era)
+paramChangeAffectsProposalsSpec =
+  describe "ParameterChange affects existing proposals" $ do
+    let largerThreshold = 51 %! 100
+        smallerThreshold = 1 %! 2
+    describe "DRep" $ do
+      let setThreshold threshold = do
+            drepVotingThresholds <- getsPParams ppDRepVotingThresholdsL
+            modifyPParams $
+              ppDRepVotingThresholdsL
+                .~ (drepVotingThresholds & dvtCommitteeNormalL .~ threshold)
+          enactThreshold threshold drepC hotCommitteeC = do
+            drepVotingThresholds <- getsPParams ppDRepVotingThresholdsL
+            let paramChange =
+                  ParameterChange
+                    SNothing
+                    ( emptyPParamsUpdate
+                        & ppuDRepVotingThresholdsL
+                          .~ SJust (drepVotingThresholds & dvtCommitteeNormalL .~ threshold)
+                    )
+                    SNothing
+            pcGai <- submitGovAction paramChange
+            submitYesVote_ (DRepVoter drepC) pcGai
+            submitYesVote_ (CommitteeVoter hotCommitteeC) pcGai
+            passNEpochs 2
+          submitTwoExampleProposalsAndVoteOnTheChild gpiCC drepKH = do
+            committeeC <- KeyHashObj <$> freshKeyHash
+            let updateCC parent = UpdateCommittee parent mempty (Map.singleton committeeC $ EpochNo 5) $ 1 %! 2
+            gai' <- submitGovAction $ updateCC $ SJust gpiCC
+            -- We submit a descendent proposal so that even though it is sufficiently
+            -- voted on, it cannot be ratified before the ParameterChange proposal
+            -- is enacted.
+            gai <- submitGovAction $ updateCC $ SJust $ GovPurposeId gai'
+            submitYesVote_ (DRepVoter $ KeyHashObj drepKH) gai
+            passEpoch -- Make the votes count do
+            pure gai
+      it "Increasing the threshold prevents a hitherto-ratifiable proposal from being ratified" $ do
+        (drepC, hotCommitteeC, gpiCC) <- electBasicCommittee
+        setThreshold smallerThreshold
+        (drepKH, _stakingKH, _paymentKP) <- setupSingleDRep 1_000_000
+        gai <- submitTwoExampleProposalsAndVoteOnTheChild gpiCC drepKH
+        isDRepAccepted gai `shouldReturn` True
+        enactThreshold largerThreshold drepC hotCommitteeC
+        isDRepAccepted gai `shouldReturn` False
```

Comment:
Does the parent govaction get enacted here? I think it'd be a good idea to assert that here also if that's the case.

Reply:
The parent-child pair of actions do not get enacted, but remain in-flight, because the parent is never voted on. This had to be done because we want to enact `ParameterChange` in between the checks. But the parent-child pair of actions can be made to be enacted finally, after the intermediate `ParameterChange` has been enacted if that will be a good test, but then this check cannot be performed before the `ParameterChange` since,

1. either the proposal has enough votes, and then a parameter-change caused it to not be enacted - in which case there has to be another condition due to which it didn't already get enacted before the parameter-change itself in that epoch
2. or, the proposal didn't have enough votes, and then a parameter-change just caused it to be enacted in the very next epoch.

Reply:
I think it'd be a good idea to also write a test to check what happens when the parent is a `ParameterChange` and gets enacted, but that makes the child proposal not accepted because the parent sets the voting thresholds to a higher value.

Reply:
I have added this test :raised_hands: Awesome suggestion @Soupstraw :rocket: